### PR TITLE
using trace from http request header

### PR DESCRIPTION
**Pull request by buggoing**

### CODE REVIEW

File: server/logger.go [hunk 1]
```diff
@@ -13,10 +14,31 @@ import (
...
+const traceHeaderKey = "traceparent"
+
 // Will absorb panics in earlier Middleware. Times the request and logs the result.
 func LogRequest(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
-		ctx, span := logtracing.StartSpan(r.Context(), fmt.Sprintf("%s %s", r.Method, r.URL.Path))
+		var opts []logtracing.StartOption
+		trace := r.Header.Get(traceHeaderKey)
+		if len(trace) > 0 {
+			ts := strings.Split(trace, "-")
```
Comment (lily1110) [inline server/logger.go:25]: What values are stored in this trace? What does each one represent?

Comment (buggoing) [reply server/logger.go:25]: https://www.w3.org/TR/trace-context/#traceparent-header
The traceparent HTTP header field identifies the incoming request in a tracing system. It has four fields:
version
trace-id
parent-id
trace-flag

**Review by sunfmin — APPROVED** [review]

**Comment by sunfmin** [comment]
最好想办法加加测试

**Comment by buggoing** [comment]
嗯，后续我加上

**Pull request merged**